liberty/build_test.go

```diff
@@ -78,10 +78,31 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 
 		Expect(result.Layers).To(HaveLen(3))
 		Expect(result.Layers[0].Name()).To(Equal("helper"))
-		Expect(result.Layers[1].Name()).To(Equal("base"))
-		Expect(result.Layers[2].Name()).To(Equal("open-liberty-runtime-full"))
+		Expect(result.Layers[1].Name()).To(Equal("open-liberty-runtime-full"))
+		Expect(result.Layers[2].Name()).To(Equal("base"))
 	})
+	
+	context("requested app server is not liberty", func() {
+		it.Before(func() {
+			Expect(os.Setenv("BP_JAVA_APP_SERVER", "notliberty")).To(Succeed())
+		})
+		
+		it.After(func() {
+			Expect(os.Unsetenv("BP_JAVA_APP_SERVER")).To(Succeed())
+		})
+		
+		it("should not run if liberty is not the requested java app server", func() {
+			ctx.Plan.Entries = []libcnb.BuildpackPlanEntry{{Name: "test"}}
 
+			buf := &bytes.Buffer{}
+			result, err := liberty.Build{Logger: bard.NewLogger(buf)}.Build(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(result.Layers).To(HaveLen(0))
+			Expect(result.Unmet).To(ContainElement(libcnb.UnmetPlanEntry{Name: "test"}))
+		})
+	})
+	
```

Comment:
```suggestion
	context("requested app server is not liberty", func() {
		it.Before(func() {
			Expect(os.Setenv("BP_JAVA_APP_SERVER", "notliberty")).To(Succeed())
		})

		it.After(func() {
			Expect(os.Unsetenv("BP_JAVA_APP_SERVER")).To(Succeed())
		})

		it("should not run if liberty is not the requested java app server", func() {
			Expect(os.MkdirAll(filepath.Join(ctx.Application.Path, "WEB-INF"), 0755)).To(Succeed())

			result, err := liberty.Build{Logger: bard.NewLogger(io.Discard)}.Build(ctx)
			Expect(err).NotTo(HaveOccurred())

			Expect(result.Layers).To(HaveLen(0))
			Expect(result.Unmet).To(HaveLen(1))
			Expect(result.Unmet[0].Name).To(Equal("liberty"))
		})
	})

```

I think this test is only coincidentally passing. The setup is not sufficient, so the build runs but doesn't contribute any layers. This should test and if you change `BP_JAVA_APP_SERVER` to `liberty` should fail.